tests/unit/Validator/RegexTest.php

```diff
@@ -0,0 +1,79 @@
+<?php
+/*
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ * This software consists of voluntary contributions made by many individuals
+ * and is licensed under the MIT license.
+ */
+namespace DocHeaderTest\Validator;
+
+use DocHeader\Validator\Regex;
+
+/**
+ * Tests for {@see \DocHeader\Validator\Regex}.
+ *
+ * @group   Unitary
+ * @author  Jefersson Nathan <malukenho@phpse.net>
+ * @license MIT
+ *
+ * @covers  \DocHeader\Validator\Regex
+ */
+final class RegexTest extends \PHPUnit_Framework_TestCase
+{
+    /**
+     * @test
+     * @dataProvider valid_regex_and_content
+     */
+    public function it_should_assert_given_regex_on_content($regex, $content)
+    {
+        $filter = new Regex($regex);
+
+        $this->assertTrue(
+            $filter->__invoke($content),
+            sprintf('"%s" does NOT match on content "%s"', $regex, $content)
+        );
+    }
+
+    /**
+     * @test
+     * @dataProvider invalid_regex_and_content
+     */
+    public function it_should_not_assert_given_regex_on_content($regex, $content)
+    {
+        $filter = new Regex($regex);
+
+        $this->assertFalse($filter->__invoke($content));
+    }
+
+    public function valid_regex_and_content()
+    {
+        return [
+            'Space around content' => ['Heya %re:\d{2}+%', '             Heya 12            '],
```

Comment:
This syntax seems a bit odd, reading it from the tests. Instead of using `%re:...%`, you could use `regexp:%...%`, which would also magically make the user rely on `%` as regex delimiters (easier escaping too)